### PR TITLE
Refactor AgentThreadPoolInstanceRegistry.

### LINKED ISSUE #1269: [AGENT] Modify the output name of the agent
## Demand advice
Modify the output name of the agent

### LINKED ISSUE #1296: [AGENT] Related Refactoring of the AgentThreadPoolInstanceRegistry Class.
## Demand advice
There are two subtasks.
- [ ] Modify the Map field of AgentThreadPoolInstanceRegistry to a static field.
- [ ] Rename the class AgentThreadPoolInstanceRegistry to AgentThreadPoolRegistry for simplicity.

### PR DESCRIPTION
**Pull request by lucca-suen**
Fixes #1296 

Changes proposed in this pull request:
- Rename the class AgentThreadPoolInstanceRegistry to AgentThreadPoolRegistry.
- Modify the holderMap and earlyConstructMap fields of the AgentThreadPoolRegistry to be static fields
- Remove the INSTANCE object and the getInstance method from the AgentThreadPoolRegistry.
- Change how the relevant classes get holderMap and earlyConstructMap.

### CODE REVIEW

**Comment by yanrongzhen** [comment]
Have you tested the agent's main flow? @lucca-suen

File: agents/threadpool/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java [hunk 1]
```diff
@@ -19,48 +19,31 @@
...
+public class AgentThreadPoolRegistry {
 
-    private static final ILog LOGGER = LogManager.getLogger(AgentThreadPoolInstanceRegistry.class);
+    private static final ILog LOGGER = LogManager.getLogger(AgentThreadPoolRegistry.class);
 
-    private final Map<String, AgentThreadPoolExecutorHolder> holderMap = new ConcurrentHashMap<>();
+    private static final Map<String, AgentThreadPoolExecutorHolder> holderMap = new ConcurrentHashMap<>();
 
-    public final Map<ThreadPoolExecutor, Class<?>> earlyConstructMap = new ConcurrentHashMap<>();
+    public static final Map<ThreadPoolExecutor, Class<?>> earlyConstructMap = new ConcurrentHashMap<>();
```
Comment (yanrongzhen) [inline agents/threadpool/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java]: Maybe you need to make this field private as well.
In addition, fields of type `static final` need to be named uppercase + underscore.

File: agents/threadpool/hippo4j-agent-plugin/thread-pool-plugin/src/main/java/cn/hippo4j/agent/plugin/thread/pool/interceptor/ThreadPoolExecutorConstructorMethodInterceptor.java [hunk 2]
```diff
@@ -52,7 +52,7 @@ public void onConstruct(EnhancedInstance objInst, Object[] allArguments) throws
         StackTraceElement declaredClassStackTraceElement = stackTraceElements.get(0);
         String declaredClassName = declaredClassStackTraceElement.getClassName();
         Class<?> declaredClass = Thread.currentThread().getContextClassLoader().loadClass(declaredClassName);
-        AgentThreadPoolInstanceRegistry.getInstance().earlyConstructMap.put((ThreadPoolExecutor) objInst, declaredClass);
+        AgentThreadPoolRegistry.earlyConstructMap.put((ThreadPoolExecutor) objInst, declaredClass);
```
Comment (yanrongzhen) [inline agents/threadpool/hippo4j-agent-plugin/thread-pool-plugin/src/main/java/cn/hippo4j/agent/plugin/thread/pool/interceptor/ThreadPoolExecutorConstructorMethodInterceptor.java]: Please change this to `getEarlyConstructMap().sth(...)`

File: agents/threadpool/hippo4j-agent-plugin/spring-plugins/spring-plugin-common/src/main/java/cn/hippo4j/agent/plugin/spring/common/support/SpringThreadPoolRegisterSupport.java [hunk 3]
```diff
@@ -47,7 +45,7 @@ public class SpringThreadPoolRegisterSupport {
     private static final Logger LOGGER = LoggerFactory.getLogger(SpringThreadPoolRegisterSupport.class);
 
     public static void registerThreadPoolInstances(ApplicationContext context) {
-        Map<ThreadPoolExecutor, Class<?>> earlyConstructMap = AgentThreadPoolInstanceRegistry.getInstance().earlyConstructMap;
+        Map<ThreadPoolExecutor, Class<?>> earlyConstructMap = AgentThreadPoolRegistry.earlyConstructMap;
```
Comment (yanrongzhen) [inline agents/threadpool/hippo4j-agent-plugin/spring-plugins/spring-plugin-common/src/main/java/cn/hippo4j/agent/plugin/spring/common/support/SpringThreadPoolRegisterSupport.java]: Please change this to getEarlyConstructMap().sth(...)

**Review by yanrongzhen — CHANGES_REQUESTED** [review]

**Comment by lucca-suen** [comment]
> Have you tested the agent's main flow? @lucca-suen

I compiled and ran hippo4j-threadpool-config-apollo-example and hippo4j-threadpool-config-apollo-spring-boot-1x-example. However, I did not find any direct or indirect dependence on the agent module in these two modules.

Maybe my process is wrong. I will reconsider the self-testing scheme in the next submission

File: agents/threadpool/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java (hunk 1 above)
Comment (lucca-suen) [reply agents/threadpool/hippo4j-agent-core/src/main/java/cn/hippo4j/agent/core/registry/AgentThreadPoolRegistry.java]: > Maybe you need to make this field private as well. In addition, fields of type `static final` need to be named uppercase + underscore.

Thank you. I'll try to reconstruct it again

**Pull request closed without merging**